packages/components/fields/radio-field/src/radio-field.story.js

```diff
@@ -0,0 +1,152 @@
+          const hintIcon = icon ? createElement(icons[icon]) : undefined;
+
+          return (
+            <RadioField
+              id={text('id', 'fruits', radioField)}
+              horizontalConstraint={select(
+                'horizontalConstraint',
+                Constraints.getAcceptedMaxPropValues(6),
+                7,
+                radioField
+              )}
+              errors={object(
+                'errors',
+                { missing: true, customError: true },
+                radioField
+              )}
+              renderError={(key) => {
+                switch (key) {
+                  case 'customError':
+                    return 'A custom error.';
+                  default:
+                    return null;
+                }
+              }}
+              isRequired={boolean('isRequired', false, radioField)}
+              touched={boolean('touched', false, radioField)}
+              name={text('name', 'fruits', radioField)}
+              value={value}
+              onChange={(event) => {
+                action('onChange')(event);
+                onChange(event.target.value);
+              }}
+              onBlur={action('onBlur')}
+              onFocus={action('onFocus')}
+              isDisabled={boolean('isDisabled', false, radioField)}
+              isReadOnly={boolean('isReadOnly', false, radioField)}
+              title={text('title', 'Fruits', radioField)}
+              hint={text('hint', 'Select an option', radioField)}
+              description={text('description', '', radioField)}
+              onInfoButtonClick={
+                boolean('show info button', false, radioField)
+                  ? action('onInfoButtonClick')
+                  : undefined
+              }
+              hintIcon={hintIcon}
+              badge={text('badge', '', radioField)}
+              direction={select(
+                'direction',
+                ['stack', 'inline'],
+                'stack',
+                radioField
+              )}
+              directionProps={{
+                scale: select(
+                  'direction props (scale)',
+                  ['xs', 's', 'm', 'l', 'xl'],
+                  'm',
+                  radioField
+                ),
+              }}
+            >
+              <RadioInput.Option
+                value={text('value #1', 'apple', radioOption1)}
+                isDisabled={boolean(
+                  'isDisabled #1 (forced)',
+                  false,
+                  radioOption1
+                )}
+                isHovered={boolean('isHovered #1', false, radioOption1)}
+                additionalContent={text('additionalContent', '', radioOption1)}
+              >
+                <Spacings.Inline scale="xs" alignItems="center">
+                  <div>{'🍎'}</div>
+                  {text('label #1', 'Apple', radioOption1)}
+                </Spacings.Inline>
```

Comment:
I think we need to wrap the text within an element for the spacing to actually works. Otherwise, the icon and the text don't have any space between them.
![image](https://user-images.githubusercontent.com/97907068/167644035-41280c48-13ac-4cc5-a5f0-509c04162848.png)

```suggestion
                <Spacings.Inline scale="xs" alignItems="center">
                  <div>{'🍎'}</div>
                  <div>{text('label #1', 'Apple', radioOption1)}</div>
                </Spacings.Inline>
```

Reply:
Maybe we could also take a look at the accessibility issues that are shown in the `Accessibility` panel in Storybook:
![image](https://user-images.githubusercontent.com/97907068/167646547-fc243c50-8395-41ff-bd91-35f72614b8b1.png)

For instance, it seems `aria-readonly` is not a supported prop for [radio elements](https://www.w3.org/TR/wai-aria-1.1/#radio).

Perhaps this can be done as a follow-up task to not increasing this one's scope.

Reply:
Sure, we will work on this as a follow-up.